OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -3136,41 +3139,7 @@ private static boolean isValidOutcomeEntry(String name) {
    }
 
    public interface OutcomeCallback {
-      void onOutcomeSuccess(String name);
-      void onOutcomeFail(int statusCode, String response);
-   }
-
-   static class OutcomeSettings {
-      private boolean cacheActive;
-
-      OutcomeSettings(Builder builder) {
-         this.cacheActive = builder.cacheActive;
-      }
-
-      boolean isCacheActive() {
-         return cacheActive;
-      }
-
-      public static class Builder {
-
-         private boolean cacheActive = true;
-
-         public static Builder newInstance() {
-            return new Builder();
-         }
-
-         private Builder() {
-         }
-
-         public Builder setCacheActive(boolean active) {
-            this.cacheActive = active;
-            return this;
-         }
-
-         public OutcomeSettings build() {
-            return new OutcomeSettings(this);
-         }
-      }
+      void onSuccess(OutcomeEvent outcomeEvent);
```

Comment:
Since we are exposing `OutcomeEvent` we need to lock down some of the methods to be `package-private`.
1. Delete `setNotificationIds` it is unused.
2. `setSession`, `setName`, `setTimestamp`.
   - Actually these shouldn't be needed at all, since they are in the constructor.
3. Constructor - make `package-private`.
4. `getParams` - make `package-private`.
   - Let's add a `getWeight`

Also `OSSessionManager.Session` needs to be a public enum.

Reply:
Sounds good, on it.